### PR TITLE
[S550268]: STK radio import

### PR DESCRIPTION
**Pull request by bryankaylor**
[S550268]: STK radio import

- Add function to get CAD nodes from a coupling
- Add function to get all components connected to another component
- Add EmitAntennaComponent, add function to get all antennas connected to a radio

### CODE REVIEW

**Comment by bryankaylor** [comment]
@maxcapodi78 Thanks, I've added unit tests and ran the style checker in this latest commit. Is there some way I can trigger the the CI/style check/Codecov above to rerun?

**Comment by MaxJPRey** [comment]
> @maxcapodi78 Thanks, I've added unit tests and ran the style checker in this latest commit. Is there some way I can trigger the the CI/style check/Codecov to rerun?

@bryankaylor They should be rerun automatically each time a new commit is made.

Otherwise, you can click on `details` :
image

and then `rerun` -on the top right corner-:
image

**Review by maxcapodi78 — APPROVED** [review]

File: _unittest/test_26_emit.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ def test_connect_components(self):
         assert connected_port is None
 
     @pytest.mark.skipif(
-        config["desktopVersion"] <= "2022.1" or is_ironpython, reason="Skipped on versions lower than 2021.2"
+        config["desktopVersion"] <= "2022.1" or is_ironpython, reason="Skipped on versions lower than 2022.2"
```
Comment (PipKat) [inline _unittest/test_26_emit.py:66]:
```suggestion
        config["desktopVersion"] <= "2022.1" or is_ironpython, reason="Skipped on versions earlier than 2022 R2."
```

File: _unittest/test_26_emit.py [hunk 2]
```diff
@@ -77,7 +77,9 @@ def test_radio_component(self):
         band.enabled = False
         assert not band.enabled
 
-    @pytest.mark.skipif(config["NonGraphical"], reason="Not functional in non-graphical mode")
+    @pytest.mark.skipif(
+        config["desktopVersion"] <= "2022.1" or is_ironpython, reason="Skipped on versions lower than 2021.2"
```
Comment (PipKat) [inline _unittest/test_26_emit.py:81]:
```suggestion
        config["desktopVersion"] <= "2022.1" or is_ironpython, reason="Skipped on versions earlier than 2021 R2."
```

File: pyaedt/emit_core/Couplings.py [hunk 3]
```diff
@@ -54,21 +54,63 @@ def coupling_names(self):
...
+
+        Returns
+        -------
+        list
+        """
         desktop_version = self._desktop.GetVersion()[0:6]
         if desktop_version >= "2022.2":
             return self._odesign.GetAvailableLinkNames()
         else:
             warnings.warn("The function linkable_design_names() requires AEDT 2022 R2 or newer.")
```
Comment (PipKat) [inline pyaedt/emit_core/Couplings.py:78]:
```suggestion
            warnings.warn("The function linkable_design_names() requires AEDT 2022 R2 or later.")
```

File: pyaedt/emit_core/Couplings.py [hunk 4]
```diff
@@ -54,21 +54,63 @@ def coupling_names(self):
...
             return []
+        
+    @property
+    def cad_nodes(self):
+        """List the cad nodes.
```
Comment (PipKat) [inline pyaedt/emit_core/Couplings.py:83]:
```suggestion
        """List the CAD nodes.
```

File: pyaedt/modeler/PrimitivesEmit.py [hunk 5]
```diff
@@ -444,6 +445,112 @@ def get_prop_nodes(self, property_filter=None):
                 filtered_nodes.append(node)
         return filtered_nodes
 
+    @pyaedt_function_handler()
+    def get_connected_components(self):
+        """Get all components that are connected (directly or indirectly) to this component.
```
Comment (PipKat) [inline pyaedt/modeler/PrimitivesEmit.py:450]:
```suggestion
        """Get all EMIT components that are connected (directly or indirectly) to this component.
```

File: pyaedt/modeler/PrimitivesEmit.py [hunk 6]
```diff
@@ -444,6 +445,112 @@ def get_prop_nodes(self, property_filter=None):
...
+
+        Args:
+            None
+
+        Returns:
+            A list containing all EmitComponents that are connected (directly or indirectly) to this component.
```
Comment (PipKat) [inline pyaedt/modeler/PrimitivesEmit.py:456]:
```suggestion
            A list containing all EMIT components that are connected to this component.
```

File: pyaedt/modeler/PrimitivesEmit.py [hunk 7]
```diff
@@ -444,6 +445,112 @@ def get_prop_nodes(self, property_filter=None):
...
+
+@EmitComponent.register_subclass("Antenna")
+class EmitAntennaComponent(EmitComponent):
+    """An Antenna component in the EMIT schematic."""
+
+    def __init__(self, components, component_name):
+        super(EmitAntennaComponent, self).__init__(components, component_name)
+
+    def get_pattern_filename(self):
+        """Get the filename of the Antenna Pattern defining this antenna
```
Comment (PipKat) [inline pyaedt/modeler/PrimitivesEmit.py:495]:
```suggestion
        """Get the filename of the antenna pattern defining this antenna.
```

File: pyaedt/modeler/PrimitivesEmit.py [hunk 8]
```diff
@@ -444,6 +445,112 @@ def get_prop_nodes(self, property_filter=None):
...
+        Args:
+
+        Returns:
+            Filename of the Antenna Pattern defining this antenna
```
Comment (PipKat) [inline pyaedt/modeler/PrimitivesEmit.py:499]:
```suggestion
            Filename of the antenna pattern.
```

File: pyaedt/modeler/PrimitivesEmit.py [hunk 9]
```diff
@@ -444,6 +445,112 @@ def get_prop_nodes(self, property_filter=None):
...
+        """
+        properties = self.get_node_properties()
+        return properties["Filename"]
+
+    def get_orientation_rpy(self):
+        """Get the RPY orientation of this Antenna.
```
Comment (PipKat) [inline pyaedt/modeler/PrimitivesEmit.py:505]:
```suggestion
        """Get the RPY orientation of this antenna.
```

File: pyaedt/modeler/PrimitivesEmit.py [hunk 10]
```diff
@@ -444,6 +445,112 @@ def get_prop_nodes(self, property_filter=None):
...
+
+        Args:
+            None
+
+        Returns:
+            A tuple containing the Roll, Pitch, and Yaw values in degrees defining this orientation
```
Comment (PipKat) [inline pyaedt/modeler/PrimitivesEmit.py:511]:
```suggestion
            Tuple containing the roll, pitch, and yaw values in degrees defining this orientation.
```

File: pyaedt/modeler/PrimitivesEmit.py [hunk 11]
```diff
@@ -444,6 +445,112 @@ def get_prop_nodes(self, property_filter=None):
...
+            return None
+
+        # Build a tuple of the orientation values.
+        parts = orientation_string.split()
+        orientation = (float(parts[0]), float(parts[1]), float(parts[2]))
+
+        return orientation
+
+    def get_position(self):
+        """Get the position of this Antenna.
```
Comment (PipKat) [inline pyaedt/modeler/PrimitivesEmit.py:533]:
```suggestion
        """Get the position of this antenna.
```

File: pyaedt/modeler/PrimitivesEmit.py [hunk 12]
```diff
@@ -444,6 +445,112 @@ def get_prop_nodes(self, property_filter=None):
...
+
+        Args:
+            None
+
+        Returns:
+            A tuple containing the X, Y, and Z offset values in meters
```
Comment (PipKat) [inline pyaedt/modeler/PrimitivesEmit.py:539]:
```suggestion
            Tuple containing the X, Y, and Z offset values in meters.
```

File: pyaedt/emit_core/Couplings.py [hunk 13]
```diff
@@ -1,22 +1,18 @@
 """
 This module contains these classes: `CouplingsEmit`.
-
 This module provides the capability to interact with EMIT Analysis & Results windows.
 """
 import warnings
 
 
 class CouplingsEmit(object):
     """Provides for interaction with the EMIT Coupling folder
```
Comment (PipKat) [inline pyaedt/emit_core/Couplings.py:9]:
```suggestion
    """Provides for interaction with the EMIT ```coupling`` folder,
```

File: pyaedt/emit_core/Couplings.py [hunk 14]
```diff
@@ -1,22 +1,18 @@
 """
 This module contains these classes: `CouplingsEmit`.
-
 This module provides the capability to interact with EMIT Analysis & Results windows.
```
Comment (PipKat) [inline pyaedt/emit_core/Couplings.py:3]:
```suggestion
This module provides for interacting with EMIT Analysis and Results windows.
```

File: pyaedt/emit_core/Couplings.py [hunk 15]
```diff
@@ -1,22 +1,18 @@
...
 """
 import warnings
 
 
 class CouplingsEmit(object):
     """Provides for interaction with the EMIT Coupling folder
-
     This class is accessible through the EMIT application results variable
     object( eg. ``emit.couplings``).
```
Comment (PipKat) [inline pyaedt/emit_core/Couplings.py:11]:
```suggestion
    object (``emit.couplings``).
```

**Pull request merged**